bpf/bpf_lxc.c

```diff
@@ -1043,6 +1043,7 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 	 */
 	if (is_defined(ENABLE_ROUTING) || hairpin_flow ||
 	    is_defined(ENABLE_HOST_ROUTING)) {
+		bool bypass = hairpin_flow && ct_status == CT_REPLY;
```

Comment:
so, this is so `hairpin_flow` are only considered for `ipv4_local_delivery(` if they are part of a REPLY request?

Reply:
Yeah that's a dirty hack atm to get it working (primarily why this PR is still WIP / in draft).

If we want the backend's `to-container` path to create the TCP_IN entry, then we can't bypass it.

Reply:
Added a comment that explains what's going on here :+1:.